src/databricks/labs/ucx/hive_metastore/data_objects.py

```diff
@@ -50,10 +56,42 @@ def _external_locations(self, tables: list[Row], mounts) -> list[ExternalLocatio
                         loc += 1
                     if not dupe:
                         external_locations.append(ExternalLocation(os.path.dirname(location) + "/"))
+                if location.startswith("jdbc"):
+                    pattern = r"(\w+)=(.*?)(?=\s*,|\s*\])"
+
+                    # Find all matches in the input string
+                    # Storage properties is of the format
+                    # "[personalAccessToken=*********(redacted), \
+                    #  httpPath=/sql/1.0/warehouses/65b52fb5bd86a7be, host=dbc-test1-aa11.cloud.databricks.com, \
+                    #  dbtable=samples.nyctaxi.trips]"
+                    matches = re.findall(pattern, table.storage_properties)
+
+                    # Create a dictionary from the matches
+                    result_dict = dict(matches)
+
+                    # Fetch the value of host from the newly created dict
+                    host = result_dict.get("host", "")
+                    port = result_dict.get("port", "")
+                    database = result_dict.get("database", "")
+                    httppath = result_dict.get("httpPath", "")
+
+                    # dbtable = result_dict.get("dbtable", "")
+
+                    # currently supporting databricks and mysql external tables
+                    # add other jdbc types
+                    if "databricks" in location.lower():
+                        jdbc_location = f"jdbc:databricks://{host};httpPath={httppath}"
+                    elif "mysql" in location.lower():
+                        jdbc_location = f"jdbc:mysql://{host}:{port}/{database}"
```

Comment:
and what happens if it's not a databricks or MySQL connection?

Reply:
I was thinking right now I can add "{location.lower()}/{host}:{port}/{database}" and we can add more database types when we get the keys in 'Storage Properties'.